backend/src/utils/cache.ts

```diff
@@ -0,0 +1,125 @@
+class CacheNode {
+    key: string;
+    value: any;
+    prev: CacheNode;
+    next: CacheNode;
+    constructor(key: string, value: any) {
+        this.key = key;
+        this.value = value;
+    }
+}
+
+export class ProductRankCache {
+    maxSize: number;
+    count: number;
+    head: CacheNode;
+    tail: CacheNode;
+    hashTable = new Map();
+    constructor(size: number) {
+        this.maxSize = size;
+        this.head = new CacheNode('head', 'head');
+        this.tail = new CacheNode('tail', 'tail');
+        this.head.next = this.tail;
+        this.tail.prev = this.head;
+        this.count = 0;
+    }
+
+    put(key: string, value: any) {
+        const node = new CacheNode(key, value);
+        this.add(node);
+        if (this.count > this.maxSize) {
+            const lowestNode = this.getlowestNode();
+            this.delete(lowestNode);
+        }
+    }
+
+    private add(node: CacheNode) {
+        let prev = this.tail.prev;
+        while (prev.value.userCount < node.value.userCount) {
+            if (prev === this.head) {
+                break;
+            }
+            prev = prev.prev;
+        }
+        const next = prev.next;
+        prev.next = node;
+        node.next = next;
+        node.prev = prev;
+        next.prev = node;
+        this.hashTable.set(node.key, node);
+        this.count++;
+    }
+
+    private getlowestNode() {
+        let node = this.tail.prev;
+        while (node.value.userCount >= node.prev.value.userCount) {
+            node = node.prev;
+        }
+        return node;
+    }
+
+    delete(node: CacheNode) {
+        const prev = node.prev;
+        const next = node.next;
```

Comment:
```suggestion
        const {prev, next} = node;
```
구조 분해 할당을 사용하셔도 될 것 같습니다!